link/tracing.go

```diff
@@ -96,20 +99,43 @@ type LSMOptions struct {
 }
 
 // attachBTFID links all BPF program types (Tracing/LSM) that they attach to a btf_id.
-func attachBTFID(program *ebpf.Program) (Link, error) {
+func attachBTFID(program *ebpf.Program, at ebpf.AttachType) (Link, error) {
 	if program.FD() < 0 {
 		return nil, fmt.Errorf("invalid program %w", sys.ErrClosedFd)
 	}
 
-	fd, err := sys.RawTracepointOpen(&sys.RawTracepointOpenAttr{
-		ProgFd: uint32(program.FD()),
-	})
-	if errors.Is(err, sys.ENOTSUPP) {
-		// This may be returned by bpf_tracing_prog_attach via bpf_arch_text_poke.
-		return nil, fmt.Errorf("create raw tracepoint: %w", ErrNotSupported)
-	}
-	if err != nil {
-		return nil, fmt.Errorf("create raw tracepoint: %w", err)
+	var fd *sys.FD
+	if err := haveBPFLinkTracing(); err != nil || at == ebpf.AttachNone {
```

Comment:
The condition to use RawTracepointOpen is hard to understand imo. Seems like this would cause AttachNone to drop into the default case at line 128 below as well?

I also prefer to not rely on feature probes to choose the implementation if possible. It's better to execute the syscall and then use feature probes if the syscall fails.

I'd refactor this into `switch at` at the top, with cases for `AttachNone` and the other types. The other types can `fallthrough` to AttachNone if the call fails.